plugins/modules/activation_key.py

```diff
@@ -275,10 +283,14 @@ def main():
             if subscriptions is not None:
                 desired_subscriptions = []
                 for subscription in subscriptions:
-                    if subscription.get('name') is not None and subscription.get('pool_id') is None:
+                    if subscription.get('name') is not None and subscription.get('pool_id') is None and subscription.get('upstream_pool_id') is None:
```

Comment:
If the required_one_of and mutually_exclusive work properly, this logic can be simplified a lot.

Reply:
that's true, I think this check pre-dates the `required_one_of` and `mutually_exclusive` above, worth checking if we can drop it